pkg/providers/vsphere/validator.go

```diff
@@ -694,3 +635,31 @@ func (v *Validator) getMissingPrivs(ctx context.Context, vsc govmomi.VSphereClie
 
 	return missingPrivs, nil
 }
+
+func (v *Validator) sameOSFamily(configs map[string]*anywherev1.VSphereMachineConfig) bool {
+	var osFamily anywherev1.OSFamily
+	for _, machineConfig := range configs {
+		if osFamily == "" {
+			osFamily = machineConfig.Spec.OSFamily
```

Comment:
Could we do this before the loop by assigning it to the first value?

Reply:
I thought the same thing, but the object that we're traversing through is a map so I don't think there's a good way to access only the first object since we don't know what the first key will be. We could pass in the `OSFamily` of the control plane machine config to use as a source of truth for this, but I think this method would be nice to reuse in our `VSphereReconciler` when we validate the machine configs so I think this is the best way to satisfy both cases

Reply:
```go
for _, machineConfig := range configs {
    osFamily = machineConfig.Spec.OSFamily
    break
}
```

Reply:
Yeah I considered doing this before. I personally don't love the idea of adding this extra loop but will add this for efficiency's sake